server/index.js

```diff
@@ -96,47 +119,79 @@ wss.on("connection", (sock, _) => {
        * }
        */
     } else if (data.messageType === "play") {
-      console.log("received play: " + JSON.stringify(data));
+      // console.log("received play: " + JSON.stringify(data));
       const room = rooms.get(data.roomId);
 
       if (!room) {
         console.error(`Invalid play. Room ${data.roomId} does not exist`);
+        return;
+      }
+
+      // If not your turn - reject play
+      if (room.currentTurn !== data.sender) {
+        sock.send(
+          JSON.stringify({
+            error: "Not your turn",
+            boardState: room.boardState, // TODO not efficient
+          })
+        );
+        return;
       }
 
-      // room.boardState.push(data.move);
+      // TODO how can we send individual plays instead?
```

Comment:
So the only way to send individual plays in my mind that doesn't involve having to duplicate the capture-algo code for the server as well as the client is to move all of the game logic to the server. In this scenario the player sends a "potential" move to the server, the server runs the algorithm and sends the new board state down to both players. You still end up sending the board state but only in one direction.

Alternatively, and perhaps slightly more complicated is that we modify the capture algorithm to keep track of the various captures that occur and only send those between the server and the client. But that would also require us to make a separate protocol for sending down capture data to both clients.

Reply:
Hmm...fair point. Moving all the logic to the server is a possibility. Does that make any other feature more difficult?

Single Connection doesn't seem to be affected
Offline is basically impossible. Intentional?
If we duplicated the logic that would, in theory, just double check the client. That overkill here?

Reply:
Resolving post discussion - can look into need for pushing to server in separate PR